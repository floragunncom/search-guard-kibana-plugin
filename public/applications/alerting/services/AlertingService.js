export default class AlertingService {
  constructor(httpClient) {
    this.httpClient = httpClient;
  }

  get(path) {
    return this.httpClient.get(path)
      .then(({ data }) => {
        if (!data.ok) throw data.resp;
        return data;
      });
  }

  delete(path) {
    return this.httpClient.delete(path)
      .then(({ data }) => {
        if (!data.ok) throw data.resp;
        return data;
      });
  }

  post(path, payload = {}) {
    return this.httpClient.post(path, payload)
      .then(({ data }) => {
        if (!data.ok) throw data.resp;
        return data;
      });
  }

  put(path, payload = {}) {
    return this.httpClient.put(path, payload)
      .then(({ data }) => {
        if (!data.ok) throw data.resp;
        return data;
      });
  }
}
