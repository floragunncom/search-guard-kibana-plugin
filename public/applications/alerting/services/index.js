/* eslint-disable @osd/eslint/require-license-header */
export { default as WatchService } from './WatchService';
export { default as AlertService } from './AlertService';
export { default as ElasticsearchService } from './ElasticsearchService';
export { default as AccountsService } from './AccountsService';
export { default as AlertingService } from './AlertingService';
export { SecurityService } from './SecurityService';
