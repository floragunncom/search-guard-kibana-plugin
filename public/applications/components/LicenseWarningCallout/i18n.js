/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { EuiI18n, EuiLink } from '@elastic/eui';
import { DOC_LINKS } from '../../utils/constants';

export const licenseCantBeLoadedText = (
  <EuiI18n
    token="sg.license_error.licenseCantBeLoaded.text"
    default="The Search Guard license information could not be loaded. Please contact your system administrator."
  />
);

export const licenseKeyIsInvalidText = (
  <>
    <EuiI18n
      token="sg.license_error.licenseKeyIsInvalid.text"
      default="The Search Guard license key is not valid for this cluster. Please contact your system administrator."
    />{' '}
    <EuiLink target="_blank" href={DOC_LINKS.LICENSING}>
      Licensing options.
    </EuiLink>
  </>
);

export const trialLicenseExpiresInText = (days) => (
  <EuiI18n
    token="sg.license_warning.trialLicenseExpiresIn.text"
    default={`Your trial license expires in ${days} days.`}
  />
);

export const licenseExpiresInText = (days) => (
  <EuiI18n
    token="sg.license_warning.licenseExpiresIn.text"
    default={`Your license expires in ${days} days.`}
  />
);
