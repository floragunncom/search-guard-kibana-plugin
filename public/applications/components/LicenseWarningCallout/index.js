/* eslint-disable @osd/eslint/require-license-header */
export { LicenseWarningCallout } from './LicenseWarningCallout';
