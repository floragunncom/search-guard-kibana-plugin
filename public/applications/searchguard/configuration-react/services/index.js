/* eslint-disable @kbn/eslint/require-license-header */
export { default as LocalStorageService } from './LocalStorageService';
export { default as ElasticsearchService } from './ElasticsearchService';

export { ApiService } from './ApiService';
export { InternalUsersService } from './InternalUsersService';
export { ActionGroupsService } from './ActionGroupsService';
export { TenantsService } from './TenantsService';
export { SgConfigService } from './SgConfigService';
export { RolesService } from './RolesService';
export { RolesMappingService } from './RolesMappingService';
export { ConfigService } from './ConfigService';
