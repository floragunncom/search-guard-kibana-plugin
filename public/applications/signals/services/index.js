/* eslint-disable @osd/eslint/require-license-header */
export { default as WatchService } from './WatchService';
export { default as AlertService } from './AlertService';
export { default as ElasticsearchService } from './ElasticsearchService';
export { default as AccountsService } from './AccountsService';
export { default as SignalsService } from './SignalsService';
export { SearchGuardService } from './SearchGuardService';
