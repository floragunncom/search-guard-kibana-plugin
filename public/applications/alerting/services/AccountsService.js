import AlertingService from './AlertingService';
import { ROUTE_PATH } from '../utils/constants';

export default class AccountsService extends AlertingService {
  constructor(httpClient, type) {
    super(httpClient);
    this.type = type;
  }

  put(account, id, type = this.type) {
    return super.put(`..${ROUTE_PATH.ACCOUNT}/${type}/${encodeURIComponent(id)}`, account);
  }

  get(id, type = this.type) {
    return super.get(`..${ROUTE_PATH.ACCOUNT}/${type}/${encodeURIComponent(id)}`);
  }

  search(query = {}) {
    return super.post(`..${ROUTE_PATH.ACCOUNTS}`, { query });
  }

  delete(id, type = this.type) {
    return super.delete(`..${ROUTE_PATH.ACCOUNT}/${type}/${encodeURIComponent(id)}`);
  }
}
