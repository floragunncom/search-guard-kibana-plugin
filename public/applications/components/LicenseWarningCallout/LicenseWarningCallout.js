/* eslint-disable @kbn/eslint/require-license-header */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { EuiCallOut, EuiText, EuiSpacer, EuiErrorBoundary } from '@elastic/eui';
import { isEmpty } from 'lodash';
import { licenseCantBeLoadedText, licenseKeyIsInvalidText, licenseExpiresInText, trialLicenseExpiresInText } from './i18n';
export function LicenseWarningCallout({ configService, errorMessage }) {
  const service = configService;

  const [licenseValid, setLicenseValid] = useState(true);
  const [error, setError] = useState(null);
  const [warning, setWarning] = useState(null);

  useState(() => {
    verifyLicense();
  }, []);

  async function verifyLicense() {
    try {
      if (isEmpty(service.get('systeminfo'))) {
        setError(licenseCantBeLoadedText);
        setLicenseValid(false);
        return;
      }

      if (!service.licenseRequired()) {
        setLicenseValid(true);
        return;
      }

      const isLicenseValid = service.licenseValid();
      setLicenseValid(isLicenseValid);

      if (errorMessage) {
        setError(errorMessage);
      } else {
        setError(licenseKeyIsInvalidText);
      }

      const isTrialLicense = service.isTrialLicense();
      const licenseExpiresInDays = service.licenseExpiresIn();

      if (isLicenseValid) {
        if (isTrialLicense && licenseExpiresInDays <= 10) {
          setWarning(trialLicenseExpiresInText(licenseExpiresInDays));
        }
        if (!isTrialLicense && service.licenseExpiresIn() <= 20) {
          setWarning(licenseExpiresInText(licenseExpiresInDays));
        }
      }
    } catch (error) {
      setLicenseValid(false);
      setError(licenseCantBeLoadedText);
      console.error('LicenseWarning - checkLicense()', error);
    }
  }

  return (
    <EuiErrorBoundary>
      {!licenseValid && (
        <>
          <EuiCallOut
            id="sg.label.licensewarning"
            data-test-subj="sg.callout.licenseWarning"
            title="Error"
            color="danger"
            iconType="alert"
          >
            <EuiText>
              <p>{error}</p>
            </EuiText>
          </EuiCallOut>
          <EuiSpacer />
        </>
      )}

      {warning && (
        <>
          <EuiCallOut
            id="sg.label.licensehint"
            data-test-subj="sg.callout.licenseHint"
            title="Warning"
            color="warning"
            iconType="help"
          >
            <EuiText>
              <p>{warning}</p>
            </EuiText>
          </EuiCallOut>
          <EuiSpacer />
        </>
      )}
    </EuiErrorBoundary>
  );
}

LicenseWarningCallout.defaultProps = {
  errorMessage: '',
};

LicenseWarningCallout.propTypes = {
  configService: PropTypes.object,
  errorMessage: PropTypes.string,
};
