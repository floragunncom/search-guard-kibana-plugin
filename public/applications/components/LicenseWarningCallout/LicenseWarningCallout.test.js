/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { LicenseWarningCallout } from './LicenseWarningCallout';
import { UiConfigService as ConfigService } from '../../../services/UiConfigService';

describe(LicenseWarningCallout.name, () => {
  let renderer;

  beforeEach(() => {
    renderer = new ShallowRenderer();
  });

  test('license cannot be loaded', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {},
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });

  test('license is required', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {
          sg_license: {
            license_required: true,
          },
        },
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });

  test('license is not required', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {
          sg_license: {
            license_required: false,
          },
        },
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });

  test('community license is valid', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {
          sg_license: {
            is_valid: true,
            license_required: false,
          },
        },
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });

  test('trial license is valid but expires in 10 days', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {
          sg_license: {
            type: 'TRIAL',
            is_valid: true,
            license_required: true,
            expiry_in_days: 10,
          },
        },
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });

  test('license is valid but expires in 20 days', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {
          sg_license: {
            is_valid: true,
            license_required: true,
            expiry_in_days: 20,
          },
        },
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });

  test('license is expired', () => {
    const configService = new ConfigService({
      config: {
        systeminfo: {
          sg_license: {
            license_required: true,
            msgs: ['License is expired'],
          },
        },
      },
    });

    const tree = renderer.render(<LicenseWarningCallout configService={configService} />);
    expect(tree).toMatchSnapshot();
  });
});
