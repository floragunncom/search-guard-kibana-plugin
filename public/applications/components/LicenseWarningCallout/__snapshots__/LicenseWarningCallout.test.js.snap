// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseWarningCallout community license is valid 1`] = `null`;

exports[`LicenseWarningCallout license cannot be loaded 1`] = `
Array [
  <div
    className="euiCallOut euiCallOut--danger"
    data-test-subj="sg.callout.licenseWarning"
    id="sg.label.licensewarning"
  >
    <div
      className="euiCallOutHeader"
    >
      <svg
        aria-hidden={true}
        className="euiIcon euiIcon--medium euiIcon-isLoading euiCallOutHeader__icon"
        focusable="false"
        height={16}
        role="img"
        style={null}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      />
      <span
        className="euiCallOutHeader__title"
      >
        Error
      </span>
    </div>
    <div
      className="euiText euiText--small"
    >
      <div
        className="euiText euiText--medium"
      >
        <p>
          The Search Guard license information could not be loaded. Please contact your system administrator.
        </p>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--l"
  />,
]
`;

exports[`LicenseWarningCallout license is not required 1`] = `null`;

exports[`LicenseWarningCallout license is required 1`] = `
Array [
  <div
    className="euiCallOut euiCallOut--danger"
    data-test-subj="sg.callout.licenseWarning"
    id="sg.label.licensewarning"
  >
    <div
      className="euiCallOutHeader"
    >
      <svg
        aria-hidden={true}
        className="euiIcon euiIcon--medium euiCallOutHeader__icon"
        focusable="false"
        height={16}
        role="img"
        style={null}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.59 10.059L7.35 5.18h1.3L8.4 10.06h-.81zm.394 1.901a.61.61 0 01-.448-.186.606.606 0 01-.186-.444c0-.174.062-.323.186-.446a.614.614 0 01.448-.184c.169 0 .315.06.44.182.124.122.186.27.186.448a.6.6 0 01-.189.446.607.607 0 01-.437.184zM2 14a1 1 0 01-.878-1.479l6-11a1 1 0 011.756 0l6 11A1 1 0 0114 14H2zm0-1h12L8 2 2 13z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="euiCallOutHeader__title"
      >
        Error
      </span>
    </div>
    <div
      className="euiText euiText--small"
    >
      <div
        className="euiText euiText--medium"
      >
        <p>
          The Search Guard license key is not valid for this cluster. Please contact your system administrator.
           
          <a
            className="euiLink euiLink--primary"
            href="https://search-guard.com/licensing/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Licensing options.
          </a>
        </p>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--l"
  />,
]
`;

exports[`LicenseWarningCallout license is valid but expires in 20 days 1`] = `
Array [
  <div
    className="euiCallOut euiCallOut--warning"
    data-test-subj="sg.callout.licenseHint"
    id="sg.label.licensehint"
  >
    <div
      className="euiCallOutHeader"
    >
      <svg
        aria-hidden={true}
        className="euiIcon euiIcon--medium euiCallOutHeader__icon"
        focusable="false"
        height={16}
        role="img"
        style={null}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.6 12.186l-1.357-1.358c-.025-.025-.058-.034-.084-.056.53-.794.84-1.746.84-2.773a4.977 4.977 0 00-.84-2.772c.026-.02.059-.03.084-.056L13.6 3.813a6.96 6.96 0 010 8.373zM8 15A6.956 6.956 0 013.814 13.6l1.358-1.358c.025-.025.034-.057.055-.084C6.02 12.688 6.974 13 8 13a4.978 4.978 0 002.773-.84c.02.026.03.058.056.083l1.357 1.358A6.956 6.956 0 018 15zm-5.601-2.813a6.963 6.963 0 010-8.373l1.359 1.358c.024.025.057.035.084.056A4.97 4.97 0 003 8c0 1.027.31 1.98.842 2.773-.027.022-.06.031-.084.056l-1.36 1.358zm5.6-.187A4 4 0 118 4a4 4 0 010 8zM8 1c1.573 0 3.019.525 4.187 1.4l-1.357 1.358c-.025.025-.035.057-.056.084A4.979 4.979 0 008 3a4.979 4.979 0 00-2.773.842c-.021-.027-.03-.059-.055-.084L3.814 2.4A6.957 6.957 0 018 1zm0-1a8.001 8.001 0 10.003 16.002A8.001 8.001 0 008 0z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="euiCallOutHeader__title"
      >
        Warning
      </span>
    </div>
    <div
      className="euiText euiText--small"
    >
      <div
        className="euiText euiText--medium"
      >
        <p>
          Your license expires in 20 days.
        </p>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--l"
  />,
]
`;

exports[`LicenseWarningCallout trial license is valid but expires in 10 days 1`] = `
Array [
  <div
    className="euiCallOut euiCallOut--warning"
    data-test-subj="sg.callout.licenseHint"
    id="sg.label.licensehint"
  >
    <div
      className="euiCallOutHeader"
    >
      <svg
        aria-hidden={true}
        className="euiIcon euiIcon--medium euiIcon-isLoading euiCallOutHeader__icon"
        focusable="false"
        height={16}
        role="img"
        style={null}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      />
      <span
        className="euiCallOutHeader__title"
      >
        Warning
      </span>
    </div>
    <div
      className="euiText euiText--small"
    >
      <div
        className="euiText euiText--medium"
      >
        <p>
          Your trial license expires in 10 days.
        </p>
      </div>
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--l"
  />,
]
`;
